src/shared/particle_dynamics/execution_unit/execution_argument.hpp

```diff
@@ -0,0 +1,89 @@
+#ifndef SPHINXSYS_EXECUTION_ARGUMENT_HPP
+#define SPHINXSYS_EXECUTION_ARGUMENT_HPP
+
+#include <sycl/sycl.hpp>
+#include <utility>
+
+#include "execution_proxy.hpp"
+
+namespace SPH::execution {
+        template<class T, sycl::access_mode access_mode>
+        class DeviceVariable {
+            using VarType = std::remove_reference_t<T>;
+        public:
+            DeviceVariable(VarType* var_addr, std::size_t var_size) : var_buffer(var_addr, var_size){}
```

Comment:
This likely can be simplified as DeviceVariable(T& var, std::size_t var_size) : var_buffer(&var, var_size){})

Reply:
On a second thought, using `T&` would require `vec.at(0)` (or `*vec.data()`) to be passed instead of `vec.data()`, in the case of vectors, which I think is less clear. Hence, I would keep `T*`.